Interactions

Axxis05 opened the pull request:
Finished interactions. Let me know if you have questions about how anything works.

Axxis05 committed:
movement to interactables

Axxis05 committed:
Finished Interactable

Finished the Interactable object type
Finished movement
Got my test scene working well
Finished right click for basic interacting

Zshuff03 commented on Assets/CameraScripts/PauseGame.cs:
```diff
@@ -9,7 +9,7 @@ public class PauseGame : MonoBehaviour {
 
 	void Update(){
-		if(Input.GetKeyDown(KeyCode.Tab)){
+		if(Input.GetKeyDown(KeyCode.Escape)){
```
Check if there is some way or need to prevent the default action of the esc key, we might need to do something like javascripts `event.preventDefault();`

Zshuff03 commented on Assets/PlayerAssets/Scripts/ClickToMove.cs:
```diff
@@ -5,21 +5,45 @@
+		playerNavMeshAgent.SetDestination(target.position);
+	}
+	///Follow the target
+	public void FollowTarget(Interactable newTarget) {
```
Just a note, we are going to have to be careful of how much we use coroutines. I would want a check to see if the character is actually a moving object then call this function if it is, not every time we right click an object. Keep that in mind.

Axxis05 replied:
Yeah i agree. Just so you know, the way the code is now, that coroutine will never run. It's there in case we decide we want to constantly follow something

Zshuff03 commented on Assets/Scripts/Interactable.cs:
```diff
@@ -0,0 +1,44 @@
+		hasInteracted = false;
+	}
+
+	void OnDrawGizmosSelected() {
```
wtf is a gizmo

Zshuff03 commented on Assets/Scripts/Interactable.cs:
```diff
@@ -0,0 +1,44 @@
+	public float radius = 3f;										//how close to interact - default 3 units
+
+	public Transform interactionTransform;
+	bool isFocus = false;
```
isFocus is kind of generic, lets rename it.

Zshuff03 commented on Assets/Scripts/Interactable.cs:
```diff
@@ -0,0 +1,44 @@
+	void Update() {
+		if(isFocus && !hasInteracted) {
+			float distance = Vector3.Distance(player.position, interactionTransform.position);
+			if(distance >= radius) {
```
dont do this programatically, use hitboxes instead. this whole check can be simplified to a couple simple steps 
```
1) user clicks thing -> set destination to the target
2) hit boxes fire no matter what, then determine if we need to move or not
3) profit with less code
```

Zshuff03 commented on Assets/TheGrid/EnvironmentScripts/HandleClicks.cs:
```diff
@@ -5,22 +5,81 @@
-			clickToMove.movePlayer(ray);
+	///Sets an interactable as the players focus and moves to it
+	void SetFocus(Interactable newFocus) {
+		if(newFocus != focus){
```
this whole focus stuff confuses me, what is focus? and what does it do? is it like the current selected thing? like in wow you click something and its hp bar and mana and buffs show up in the top?

Axxis05 replied:
a focus is basically the target of the action. its what you're "focusing on" with your interactions

Zshuff03 commented on ProjectSettings/TagManager.asset:
```diff
@@ -18,8 +18,8 @@ TagManager:
   - 
-  - 
-  - 
+  - Interactable
```
nice conflicts in this file are easily solvable if there are any. didnt realize it was literally just a text file for the most part

Axxis05 committed:
Added comments

Made things more readable with comments to define what each function and variable does
